### PR TITLE
master

### PR DESCRIPTION
**Pull request by kasperosterbye**
Added GeneratorWrapper to incorporate class Generator into the Iterator framework

Also added tests for the class. 
I am a bit uncertain if the class should be split into two - the wrapper and the decorator.
It is not clear what the extension methods on Block should be called, so they are not there at the moment.

### CODE REVIEW

File: src/Iterators/GeneratorWrapper.class.st [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	^next
+]
+
+{ #category : #accessing }
+GeneratorWrapper >> peek [
+	hasNext ifFalse: [ IteratorIsAtEnd signal].
+	^peek
+]
+
+{ #category : #'as yet unclassified' }
```
Comment (juliendelplanque) [inline src/Iterators/GeneratorWrapper.class.st]: Should be classified.

File: src/Iterators/GeneratorWrapper.class.st [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	],
+	#category : #'Iterators-Wrappers'
+}
+
+{ #category : #'instance creation' }
+GeneratorWrapper class >> generator: generatorBlock [
+	^self new generator: generatorBlock 
+]
+
+{ #category : #'as yet unclassified' }
```
Comment (juliendelplanque) [inline src/Iterators/GeneratorWrapper.class.st]: Should be classified.

File: src/Iterators/GeneratorWrapper.class.st [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+GeneratorWrapper class >> pipe: block [
+	^self new pipe: block
+]
+
+{ #category : #'as yet unclassified' }
```
Comment (juliendelplanque) [inline src/Iterators/GeneratorWrapper.class.st]: Should be classified

File: src/Iterators-Tests/GeneratorWrapperTest.class.st [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	| res |
+	res := (GeneratorWrapper generator: [:gen | 1 to: 5 do:[:i | gen yield: i*2]]) > Array.
+	self assert: res equals: #(2 4 6 8 10)
+]
+
+{ #category : #tests }
+GeneratorWrapperTest >> testPipingGenerator [
+	| res |
+	res := ((1 to: 3) iterator
+		|
```
Comment (juliendelplanque) [inline src/Iterators-Tests/GeneratorWrapperTest.class.st:28]: I would like that the `|` is not used in the kernel of Iterators, this is part of the DSL that can be changed.
I recommand to use `#decoratedBy:` instead.

**Review by juliendelplanque — CHANGES_REQUESTED** [review]
Thanks for your contribution kasper!

once the requests will be addressed, I'll merge these changes

**Comment by kasperosterbye** [comment]
I'll make a new PR and close this one.

**Comment by juliendelplanque** [comment]
Super! Thanks for your energy :-)

**Pull request closed without merging**